synapse/rest/client/v1/room.py

```diff
@@ -449,30 +473,40 @@ async def on_POST(self, request, room_id):
 
         events_to_create = body["events"]
 
-        # If provided, connect the chunk to the last insertion point
-        # The chunk ID passed in comes from the chunk_id in the
-        # "insertion" event from the previous chunk.
+        # Figure out which chunk to connect to. If they passed in
+        # chunk_id_from_query let's use it. The chunk ID passed in comes
+        # from the chunk_id in the "insertion" event from the previous chunk.
+        last_event_in_chunk = events_to_create[-1]
+        chunk_id_to_connect_to = chunk_id_from_query
         if chunk_id_from_query:
-            last_event_in_chunk = events_to_create[-1]
-            last_event_in_chunk["content"][
-                EventContentFields.MSC2716_CHUNK_ID
-            ] = chunk_id_from_query
+            # TODO: Verify the chunk_id_from_query corresponds to an insertion event
+            pass
+        # Otherwise, create an insertion event to be based off of and connect
+        # to as a starting point.
```

Comment:
```suggestion
        # We don't always have an insertion event to hang more history off of
        # (ideally they'd be one in the main DAG, but that's not the case if
        # we're wanting to add history to e.g. existing rooms without such an
        # event), in which case we just create a new insertion event that will
        # then get pointed to by a "marker" event.
```

Or something to that effect, I'm terrible with words.